src/checkqueue.h

```diff
@@ -114,6 +124,12 @@ class CCheckQueue
                     if (fMaster && nTodo == 0) {
                         // All checks are done; master performs batch verification
                         if constexpr (std::is_same_v<R, ScriptFailureResult>) {
+                            for (auto& bucket : m_signature_buckets) {
+                                for (const auto& sig : bucket) {
+                                    m_batch.Add(sig.sig, sig.pubkey, sig.sighash);
```

Comment:
https://github.com/bitcoin/bitcoin/pull/29491/commits/6055f3a6f530818e2805d11debb1fb2ce81fc5c0: This leaves the master thread to do all the "Pubkey recovery"(I think that's what its called?) work in each `batch.Add` call, which is substantial. We can call "secp256k1_xonly_pubkey_parse" when collecting the signatures so the master thread is not stuck doing all the work to parse the XOnlyPubkey.